iroh-rpc-client/src/network.rs

```diff
@@ -270,23 +271,39 @@ impl P2pClient {
     }
 
     #[tracing::instrument(skip(self))]
-    pub async fn check(&self) -> StatusRow {
-        let status: ServiceStatus = self
-            .version()
-            .await
-            .map(|_| ServiceStatus::Serving)
-            .unwrap_or_else(|_e| ServiceStatus::Unknown);
-        StatusRow {
+    pub async fn check(&self) -> ServiceStatus {
+        let (status, version) = match self.version().await {
+            Ok(version) => (StatusType::Serving, version),
+            Err(_) => (StatusType::Down, String::new()),
+        };
+        ServiceStatus {
             name: "p2p",
-            number: 2,
+            number: 1,
             status,
+            version,
         }
     }
 
     #[tracing::instrument(skip(self))]
-    pub async fn watch(&self) -> impl Stream<Item = StatusRow> {
-        // todo
-        futures::stream::pending()
+    pub async fn watch(&self) -> impl Stream<Item = ServiceStatus> {
+        let client = self.client.clone();
+        stream! {
+            loop {
+                let res = client.server_streaming(WatchRequest).await;
+                match res {
+                    Ok(mut res) => {
+                        while let Some(v) = res.next().await {
```

Comment:
while let Some(Ok(...)) =

Reply:
that would break the loop on an error, which is not what happens here